mesa/space.py

```diff
@@ -479,9 +474,9 @@ def move_to_empty(
                     break
         else:
             new_pos = agent.random.choice(sorted(self.empties))
+        self.remove_agent(agent)
```

Comment:
Caution when reviewing this. I intentionally move this line to be before the `_place_agent`.